### DISCUSSION TITLE
保存到 Telegraph

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#2795] [插件说明]
描述
---

> 导出阅读模式的内容到 Telegra.ph，[请看示例](https://telegra.ph/%E7%9C%8B%E6%9D%BF%E5%8F%8C%E5%90%91%E9%93%BE%E6%8E%A5%E7%9F%A5%E8%AF%86%E5%9B%BE%E8%B0%B1-%E7%AE%80%E6%82%A6-2-2-%E6%83%B3%E8%AE%A9%E4%BD%A0%E7%9A%84%E7%A8%8D%E5%90%8E%E8%AF%BB%E4%B8%8D%E5%86%8D%E4%B8%8D%E8%AF%BB-09-07)。

> 此插件由简悦社区用户 [@OverflowCat](http://about.overflow.cat/) 提供，非常感谢他的付出 🙏

地址
---

[https://simpread.ksria.cn/plugins/details/eqbLyHYkN4](https://simpread.ksria.cn/plugins/details/eqbLyHYkN4)

配置（选项页）
---

> 当首次安装时会自动弹出设置界面

![image](https://user-images.githubusercontent.com/81074/132287892-34f207e2-9ec0-4d91-bb5a-3d3276b18329.png)

### 申请新的 Token

> 此功能**内置了一个 Token，但请勿使用此方式**，请自己申请一个 Token，很简单无需其它操作，只需要点下 **生成新 Token**。

![image](https://user-images.githubusercontent.com/81074/132288455-ebd89565-dc56-410b-8549-ee76e2badde5.png)

> Telegra.ph 支持多组不同的 Token，你可以在这里切换不同的 Token（每个 Token 都拥有不同的文章列表，类似账户一样）

![image](https://user-images.githubusercontent.com/81074/132288823-70a9dd7a-fca6-40cf-b139-4e43d8158705.png)

> 如果是新生成的，请选择**最新生成的一个**。

![image](https://user-images.githubusercontent.com/81074/132288908-7d862973-7c4f-4c5d-b5fb-f109c60d340b.png)

### 图床

> Telegra.ph 有自己的图床，有一定动手能力的话，可以通过 https://github.com/OverflowCat/telegraph-uploader 来自建，否则请不要改动这里。

![image](https://user-images.githubusercontent.com/81074/132289033-f0720f50-db20-4b8f-bd1e-8976714fc95a.png)

> 开启后类似 [Notion 图床功能](http://ksria.com/simpread/docs/#/Notion?id=%E5%9B%BE%E5%BA%8A)。

![image](https://user-images.githubusercontent.com/81074/132289627-04f5b0af-8c6b-4919-a565-e4e26032bf22.png)

### 分享服务

> 内置了 Telegra.ph 四种服务。

![image](https://user-images.githubusercontent.com/81074/132289065-16120126-52bf-4188-b8ad-ebde82b395ff.png)

### 其它选项

> 除上述功能外，还提供了一些其它个性化设置。

![image](https://user-images.githubusercontent.com/81074/132289180-6c3830ec-1968-4430-9756-f513d5455002.png)

## 如何使用

> 内置了 `t g`  通过触发快捷键使用。

> 阅读模式 → 右下角 → 动作 → 插件触发器

![image](https://user-images.githubusercontent.com/81074/132289352-783e9752-0e46-4e5a-a3c5-d61c5f8fddc6.png)

更新日志
---

### 1.0.0

- 图床，并可自建

- 生成 Token，并可选择不同的 Token

- 分享服务，包括：[telegra.ph](http://telegra.ph/) [te.legra.ph](http://te.legra.ph/) [graph.org](http://graph.org/) [tategaki.de](http://tategaki.de/)

- 生成链接后，是否复制到剪切板

- 生成链接后，是否直接打开

- 快捷键 `t g` 支持直接发送

关联
---

- [当加入稍后读后，自动发送到你的 Telegram Channel](https://github.com/Kenshin/simpread/discussions/3092)

- [将当前阅读模式转换为 Telegraph Page 并发送到你的 Telegram Channel](https://github.com/Kenshin/simpread/discussions/3337)

- [适配了一组关于 Telegram 的新站](https://github.com/Kenshin/simpread/discussions/2883)

引申
---

> 安装插件或带有插件触发器的插件失效的 [解决方案](https://github.com/Kenshin/simpread/discussions/2342)。

**Comment by hellozgl** [comment]
请教：在telegraph插件的设置界面，不小心多点了几下生成了多个不同的token，这些token就一直保存在下拉的菜单中了，（强迫症）请问有否办法把多出的token从下拉菜单中删除呢？以免以后不小心下拉选错token

**Reply by hellozgl** [reply]
已自行解决。修改配置文件删除相关token即可

**Comment by hellozgl** [comment]
请教一个telegraph的使用问题。如果利用“保存到telegraph”功能，将文章保存到telegra.ph上，日后该如何再找到这篇文章呢，telegraph没有账户体系，只能通过保存网址的方式来追溯保存过的文章吗

**Reply by Kenshin** [reply]
通过 token 可以找到全部的文章，原计划是应该有个管理功能。

单纯的生成  Telegraph 意义并不大，这个功能的真正形态可以看这个 https://github.com/Kenshin/simpread/discussions/3092
